### DISCUSSION TITLE
[AI-Deck] Any insights on cfloader flash img hangs at 99% specifically?

### DISCUSSION THREAD

**Discussion by chengliu-LR** [bitcraze/discussions#369] [Q&A] [answered]
Hi all,

I am using cfloader to flash some `target.board.devices.flash.img` to AI-Deck. For typical `img`s like the `wifi-demo` and `pulp-dronet`, it works perfectly. However, when it comes to the `classification-demo` and my own compiled programs, the cfloader hangs at `\ 99% Writing to bcAI:gap8 deck memory` everytime. Do you know possible reasons for this?

Thank you!

**Comment by evoggy** [comment]
Hi! Apologies for the late reply in the other thread. Any chance you could share the binary that you've built for the classification demo and I can try to flash it and debug it?

**Reply by chengliu-LR** [reply]
Dear Marcus,

Thank you for the reply! Definitely I can share the binary.
I uploaded the classification demo binary to google drive and you can download through this link:
https://drive.google.com/file/d/1Zt3vwltn3qcVWid420w8kIjEGiLqbzzm/view?usp=sharing
Thank you for the help!

**Comment by knmcguire** [comment]
I can reproduce this error actually with the release 2022.09. 

Could you share the content of the cfclient 'help'>'about'>'debug' window? So then we know which version firmware your crazyflie got

**Reply by chengliu-LR** [reply]
Hi Kimberly,
Thank you for the help! This is the firmware version I used: 
`Firmware: 5c4261e5a9ff (CLEAN)`
The error can be reproduced for release 2022.09 and 2022.05 both from my side.

**Comment by knmcguire** [comment]
I've made an issue here, as this seems to be a problem to investigate and change something: https://github.com/bitcraze/aideck-gap8-examples/issues/104

**Comment by krichardsson** [comment, marked as answer]
Hopefully this problem has been solved in PR https://github.com/bitcraze/crazyflie-firmware/pull/1162.
@chengliu-LR it would be awesome if you could try it out and see if works for you. To test it pull the latest code from [crazyflie-firmware](https://github.com/bitcraze/crazyflie-firmware), build and flash to the Crazyflie. When this is done, the 99% problem of GAP8 flashing should be fixed.

**Reply by chengliu-LR** [reply]
Dear Kristoffer,

Thank you for fixing this! I already tested this and the cfloader works totally well with the classification demo and my own binary file now.
